sql/builtin_functions.cc

```diff
@@ -3439,6 +3440,54 @@ bind(const std::vector<BoundSqlExpression> & args,
     }
 }
 
+
+BoundFunction get_mime_type(const std::vector<BoundSqlExpression> & args)
+{
+    checkArgsSize(args.size(), 1);
+    auto outputInfo = std::make_shared<Utf8StringValueInfo>();
+
+    return {[=] (const std::vector<ExpressionValue> & args,
+                 const SqlRowScope & scope) -> ExpressionValue
+            {
+                const char *mime;
+                magic_t magic;
+
+                magic = magic_open(MAGIC_NONE);
+                if(magic == NULL) {
+                    throw ML::Exception("Unable to initialize the magic library");
+                }
+
+                if(magic_load(magic, NULL) != 0) {
+                    magic_close(magic);
+                    throw ML::Exception("Error loading magic database - %s\n", magic_error(magic));
+                }
+
+                if(!args[0].isAtom())
+                    throw MLDB::Exception("Mime type extraction requires that an atomic value "
+                            "of type BLOB is passed to it. Use the fetcher function to get the file blob.");
```

Comment:
Should we be as explicit as that? (Use the fetcher function... which may be renamed and is not the only way of doing it.)
If we want to be that explicit, then we could say `fetcher()[content]`